README.md

```diff
@@ -1,2 +1,319 @@
+each of which needs its own styling and tweaks to fit in to the application.
+If instead libraries and components all use the standard toast,
+the application developer can centrally style and coordinate them.
+
+
+## Sample code
+
+The standard toast can be used according to two different patterns.
+
+The first defines a `<std-toast>` HTML element,
+then shows it with configurations via a method on the element.
+This can be used to declaratively predefine toasts the application will need,
+and then show them inside the application logic.
+
+```html
+<script type="module">
+import 'std:elements/toast';
+</script>
+
+<std-toast id="sample-toast" theme="info">
+    Hello World!
+</std-toast>
+```
+
+```js
+document.querySelector('#sample-toast').show({
+    duration: 3000
+});
+```
+
+The second imports the `showToast()` function from the `"std:elements/toast"` module,
+which takes in a message and some configurations and creates and shows a toast in the DOM.
+This is more convenient for one-off toasts,
+or for JavaScript-driven situations,
+similar to how `alert()` is currently used.
+
+```js
+import { showToast } from 'std:elements/toast';
+
+const toast = showToast("Hello World!", {
+    theme: "info",
+    duration: 3000
+});
+```
+
+## Goals
+
+Across popular toast implementations there are recurring patterns,
+which the standard toast aims to accomplish natively.
+
+- The component will be accessible by default;
+  native accessibility is is a strong priority for toasts,
+  as they can be difficult to make properly accessible.
+
+- Toast implementations are often shaped similarly,
+  and a goal of the standard toast is to make it as easy as possible for developers
+  to build and style toasts that conform to those common shapes.
+
+    - Toasts almost always support a message, action, and close button.
+
+    - Toasts often support an icon and a title.
+
+- The positioning of the toast must be intuitive,
+  so the standard toast will come with built-in support for common positions,
+  as well as a sensible default.
+
+- To balance ease of use with customization,
+  the standard toast will support creating and showing with one JavaScript function,
+  as well as writing a custom view with a `<std-toast>` element
+  and showing that with a method.
+
+- The standard toast will come with support for showing multiple toasts,
+  either by stacking them in the view,
+  or queueing them and displaying sequentially.
+
+## Proposed API
+
+The element is provided as a [built-in module](https://github.com/tc39/proposal-javascript-standard-library/blob/master/README.md),
+named `"std:elements/toast"`.
+
+### The `<std-toast>` element
+
+#### Attributes
+
+- [Global attributes](https://html.spec.whatwg.org/multipage/dom.html#global-attributes)
+- `open`: a boolean attribute, determining whether the toast is visible or not (according to the default styles). By default toasts are not shown.
```

Comment:
The idea is that, like `<dialog>`, the default styling for `std-toast:not([open])` is `display: none`. I guess this could be made explicit in the default styling section, which is currently just a TODO.

Reply:
If it's following a proven pattern, that's good.